application-analytics-ui/src/main/resources/Analytics/Code/Macros/Filter.xml

```diff
@@ -0,0 +1,507 @@
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </id>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>2</number>
+        <prettyName>Macro name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <priority>
+        <disabled>0</disabled>
+        <name>priority</name>
+        <number>11</number>
+        <numberType>integer</numberType>
+        <prettyName>Priority</prettyName>
+        <size>10</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.NumberClass</classType>
+      </priority>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context>
+        <value>wiki</value>
+      </async_context>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+#macro(executeMacro)
+#set ($discard =  $xwiki.ssx.use('Analytics.Code.Macros.Filter'))
+#set ($placeholderStart = '2007-01-01')
+#set ($placeholderEnd = $datetool.get('yyyy-MM-dd'))
+#if ($request.analyticsStartDate &amp;&amp; $request.analyticsStartDate != '')
+  #set ($placeholderStart = $request.analyticsStartDate)
+#end
+#if ($request.analyticsEndDate &amp;&amp; $request.analyticsEndDate != '')
+  #set ($placeholderEnd = $request.analyticsEndDate)
+#end
+{{html clean="false"}}
+&lt;div class="xform half"&gt;
+	&lt;form id="analyticsFilter" class="form-flex"&gt;
+		&lt;div class="form-item flex-item"&gt;
+			&lt;!-- Start Date Label &amp; Input Group --&gt;
+			&lt;label for="analyticsStartDate"&gt;$services.localization.render('rendering.macro.analyticsFilter.startDate')&lt;/label&gt;
```

Comment:
Because translations can be added by users, besides the ones that come protected by a license, you need to make sure that they don't introduce security issues. So always wrap the `$services.localization.render` in a `$escapetool.xml` (check the pages for where you need this)

Reply:
Should I replace all the instances of $services,localization.render with a regex just to be safe?